apps/rush-lib/src/logic/deploy/DeployManager.ts

```diff
@@ -181,6 +181,12 @@ export class DeployManager {
           throw new Error(`Error resolving ${dependencyPackageName} from ${packageJsonRealFolderPath}`);
         }
 
+        if (resolvedDependency === dependencyPackageName) {
+          // When resolve.sync() encounters a system package such as "fs" or "path" (detected via the isCore()
+          // function), it simply returns its original input, instead of a proper file path.
+          continue;
+        }
+
```

Comment:
We should not just skip it, we need another strategy to find path to that dependency

Reply:
@arikon Could you open an issue or PR to fix the [resolve](https://github.com/browserify/resolve) library? The maintainer is usually pretty fast about responding.

Reply:
@octogonz I've added a comment here https://github.com/browserify/resolve/issues/216#issuecomment-636434580

Reply:
@octogonz So we could use `isCore()` to check if module is system, and add `/` to it's name before calling `resolve.sync()`

https://github.com/browserify/resolve/blob/master/lib/is-core.js

Reply:
This patch works for me:

```diff
diff --git a/apps/rush-lib/src/logic/deploy/DeployManager.ts b/apps/rush-lib/src/logic/deploy/DeployManager.ts
index e4d4a290b..a205defe0 100644
--- a/apps/rush-lib/src/logic/deploy/DeployManager.ts
+++ b/apps/rush-lib/src/logic/deploy/DeployManager.ts
@@ -206,7 +206,8 @@ export class DeployManager {
       }

       for (const dependencyPackageName of allDependencyNames) {
-        const resolvedDependency: string = resolve.sync(dependencyPackageName, {
+        const dependencyPackageNameToResolve: string = dependencyPackageName + (resolve.isCore(dependencyPackageName) ? '/' : '');
+        const resolvedDependency: string = resolve.sync(dependencyPackageNameToResolve, {
           basedir: packageJsonRealFolderPath,
           preserveSymlinks: false,
           packageFilter: (pkg, dir) => {
```

Reply:
Thanks! 🙏